unittests/test_modules.py

```diff
@@ -402,3 +402,45 @@ def test_mapping_deep_cycle(self):
         }
         self.assertRaisesRegex(EnvironError, 'm0->m1->m3->m4->m2->m1',
                                self.modules_system.load_module, 'm0')
+
+    def test_mapping_from_file_simple(self):
+        with open('module_mappings_file.txt', 'w') as f:
+            print('m1:m2, m3'
+                  '\nm2 : m4',
+                  '\n  #m5: m6',
+                  '\n',
+                  '\nm1: m7, # An inline comment after the coma',
+                  '\nm1: m7 # An inline comment with no coma',
+                  file=f)
+        reference_map = {
+            'm1': ['m2', 'm3', 'm7', 'm7'],
```

Comment:
That's an interesting case. I think the loader should make sure to remove duplicates from the lists.